src/lib/expression/expression_utils.cpp

```diff
@@ -108,15 +108,13 @@ void expression_adapt_to_different_lqp(std::shared_ptr<AbstractExpression>& expr
 
 std::shared_ptr<LQPColumnExpression> expression_adapt_to_different_lqp(const LQPColumnExpression& lqp_column_expression,
                                                                        const LQPNodeMapping& node_mapping) {
-  const auto node = lqp_column_expression.column_reference.original_node();
+  const auto node = lqp_column_expression.original_node.lock();
+  Assert(node, "LQPColumnExpression is expired");
```

Comment:
Just for understanding: in which cases/how can `LQPColumnExpressions` be expired?

Reply:
> If the original_node is not referenced by any shared_ptr anymore, it is deleted and the weak_ptr is expired. This should never be the case. If `original_node.lock()` is `nullptr`, the LQP is defective.

Would that help as a comment in lqp_column_expression.hpp?

Reply:
It's basically the nature of a weak pointer and it's ´lock´ function, however, I think it helps.